Numpy > 1.20

MalteOlle opened the pull request:
Calling `pg.functions.colorDistance([])` raises an error when using numpy >= 1.20, because `np.float` is deprecated, see https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations.

- Fixes #2738 
- Adds MaD GUI as package, that uses pyqtgraph (README)

### Other Tasks 

<details>
  <summary>Bump Dependency Versions</summary>

### Files that need updates
    
Confirm the following files have been either updated or there has been a determination that no update is needed.

- [x] `README.md`
- [ ] ~`setup.py`~
- [ ] ~`tox.ini`~
- [ ] ~`.github/workflows/main.yml` and associated `requirements.txt` and conda `environemt.yml` files~
- [ ] ~`pyproject.toml`~
- [ ] ~`binder/requirements.txt`~

</details>

<details>
    <summary>Pre-Release Checklist</summary>

### Pre Release Checklist

- [ ] Update version info in `__init__.py` 
- [ ] Update `CHANGELOG` primarily using contents from automated changelog generation in GitHub release page
- [ ] Have git tag in the format of pyqtgraph-<version>

</details>


<details>
  <summary>Post-Release Checklist</summary>

### Steps To Complete

- [ ] Append `.dev0` to `__version__` in `__init__.py`
- [ ] Announce on mail list
- [ ] Announce on Twitter

</details>

j9ac9k commented:
Whoops, nice catch on the use of the deprecated API.  We should add a test to cover that codepath, but that's outside the scope of this PR.

Why is the test being removed?  Also, nice library, thanks for adding to the README!

MalteOlle commented:
> Whoops, nice catch on the use of the deprecated API. We should add a test to cover that codepath, but that's outside the scope of this PR.
> 
> Why is the test being removed? Also, nice library, thanks for adding to the README!

Ah I just removed a test which I had added in the previous commit, but which did not turn out to really test the issue I thought (wanted to test colorDistance by creating a colormap with linearize=True, but wasn't able to get to the colorDistance function this way.

So in the end, I simply added a "stupid" test, that just checks whether colorDistance returns without raising am error If number of colors is 0 and 2.

j9ac9k commented:
Whops, should have noticed that :D  Thanks for the PR!
